src/editor/plugins/lsp/kcl/index.ts

```diff
@@ -1,247 +1,194 @@
 import type { Extension } from '@codemirror/state'
-import type { EditorView, PluginValue, ViewUpdate } from '@codemirror/view'
-import { ViewPlugin } from '@codemirror/view'
+import { EditorView, ViewPlugin } from '@codemirror/view'
 import type {
   LanguageServerClient,
   LanguageServerOptions,
 } from '@kittycad/codemirror-lsp-client'
-import {
-  lspCodeActionEvent,
-  lspFormatCodeEvent,
-  lspPlugin,
-  lspRenameEvent,
-} from '@kittycad/codemirror-lsp-client'
+import { lspPlugin } from '@kittycad/codemirror-lsp-client'
 import {
   updateOutsideEditorEvent,
   editorCodeUpdateEvent,
   type KclManager,
 } from '@src/lang/KclManager'
-import { kclManager, rustContext } from '@src/lib/singletons'
-import { deferExecution } from '@src/lib/utils'
-import { jsAppSettings } from '@src/lib/settings/settingsUtils'
-
-import type { UpdateCanExecuteParams } from '@rust/kcl-lib/bindings/UpdateCanExecuteParams'
-import type { UpdateCanExecuteResponse } from '@rust/kcl-lib/bindings/UpdateCanExecuteResponse'
-import type { UpdateUnitsParams } from '@rust/kcl-lib/bindings/UpdateUnitsParams'
-import type { UpdateUnitsResponse } from '@rust/kcl-lib/bindings/UpdateUnitsResponse'
 
 import { copilotPluginEvent } from '@src/editor/plugins/lsp/copilot'
 import { processCodeMirrorRanges } from '@src/lib/selections'
+import { deferExecution } from '@src/lib/utils'
 import type {
   SceneGraphDelta,
   SourceDelta,
 } from '@rust/kcl-lib/bindings/FrontendApi'
-import type { SceneEntities } from '@src/clientSideScene/sceneEntities'
+import { jsAppSettings } from '@src/lib/settings/settingsUtils'
 
+/** Debounce for execution in ms */
 const changesDelay = 600
 
-/** A view plugin that requests completions from the server after a delay */
-export class KclPlugin implements PluginValue {
-  private viewUpdate: ViewUpdate | null = null
-  private client: LanguageServerClient
-  private readonly kclManager: KclManager
-  private readonly sceneEntitiesManager: SceneEntities
-
-  constructor(
-    client: LanguageServerClient,
-    view: EditorView,
-    systemDeps: {
-      kclManager: KclManager
-      sceneEntitiesManager: SceneEntities
-    }
-  ) {
-    this.client = client
-    this.kclManager = systemDeps.kclManager
-    this.sceneEntitiesManager = systemDeps.sceneEntitiesManager
-
-    // Gotcha: Code can be written into the CodeMirror editor but not propagated to kclManager.code
-    // because the update function has not run. We need to initialize the kclManager.code when lsp initializes
-    // because new code could have been written into the editor before the update callback is initialized.
-    // There appears to be limited ways to safely get the current doc content. This appears to be sync and safe.
-    const kclLspPlugin = this.client.plugins.find((plugin) => {
-      return plugin.client.name === 'kcl'
-    })
-    if (kclLspPlugin) {
-      systemDeps.kclManager.code = view.state.doc.toString()
-    }
```

Comment:
I revived this block of code as `hackSetInitialCodeFromLsp`, a "run once" extension that sets the code on kclManager initially. I believe we still need it until we get rid of React / tearing down the EditorView, although we have no automated tests that capture that.